integration-tests/main_test.go

```diff
@@ -134,6 +137,14 @@ func TestIntegration(t *testing.T) {
 		Proxy: &api.AgentServiceConnectProxyConfig{
 			DestinationServiceName: "backend",
 			LocalServicePort:       8080,
+			Upstreams: []api.Upstream{
+				{
+					DestinationType:  api.UpstreamDestTypeService,
+					DestinationName:  "frontend",
+					LocalBindPort:    upstreamLocalBindPort.Int(),
+					LocalBindAddress: "0.0.0.0",
+				},
```

Comment:
Q: Just confirming, we configure an upstream cycle (frontend and backend have each other as an upstream) in order to validate graceful shutdown when the proxy has both inbound and outbound connections?

Reply:
Yep exactly - the basic setup was sufficient for testing that _outbound_ requests from the client to server were able to succeed during a configured shutdown grace period, but I wanted to add test coverage to ensure that _inbound_ connection (from server to client) are rejected (if `-shutdown-drain-listeners` is configured) while the client is gracefully shutting down.